src/charm.py

```diff
@@ -903,7 +903,7 @@ def _create_pgdata(self, container: Container):
         """Create the PostgreSQL data directory."""
         if not container.exists(self.pgdata_path):
             container.make_dir(
-                self.pgdata_path, permissions=0o770, user=WORKLOAD_OS_USER, group=WORKLOAD_OS_GROUP
```

Comment:
For the history, it has been noticed on Sept 23 while working on Juju 3.6 beta testing, but left unrecognised in daily routine: https://github.com/canonical/postgresql-k8s-operator/pull/689/commits/52271e160f044a0a2647fbffd96279fab89b08b1

TIL!